File: notebooks/libs/llm/chatgpt.py
import os
import httpx

from langchain_openai import ChatOpenAI
from getpass import getpass

if os.environ.get('OPENAI_API_KEY') is None:
    os.environ['OPENAI_API_KEY'] = getpass('请输入你的openai key: ')

chatgpt = ChatOpenAI(
    model="gpt-3.5-turbo",
    temperature=0.8,
    api_key=os.environ['OPENAI_API_KEY'],
    http_client=httpx.Client(proxy='http://localhost:7890')
)

File: javascript/libs/llm/ChatGPT.ts
import {ChatOpenAI} from "@langchain/openai";
import {createInterface} from 'readline'
import {HttpsProxyAgent} from 'https-proxy-agent'

if (!process.env.OPENAI_API_KEY) {
    const readline = createInterface({
        input: process.stdin,
        output: process.stdout,
    });
    readline.question('请输入你的openai key: ', key => {
        process.env.OPENAI_API_KEY = key;
        readline.close();
    })

}
const llm = new ChatOpenAI({
    temperature: 0.8,
    apiKey: process.env.OPENAI_API_KEY,
    configuration: {
        httpAgent: new HttpsProxyAgent('http://localhost:7890'),
    }
});
export default llm